src/middleware/node/middleware.ts

```diff
@@ -93,13 +98,30 @@ export async function middleware(
   }, 9000).unref();
 
   try {
-    const payload = await getPayload(request);
+    let payload: Buffer;
+
+    if ("body" in request) {
+      if (
+        typeof request.body === "object" &&
+        "rawBody" in request &&
+        request.rawBody instanceof Buffer
+      ) {
+        // The body is already an Object and rawBody is a Buffer (e.g. GCF)
+        payload = request.rawBody;
```

Comment:
If we are sending the raw body as a Buffer from GCF, then why do we need the JSON obect in a new `body` parameter?

Reply:
There is no new body parameter. I removed it to satisfy this for now. We can discuss this when I have more time ;)